index.ts

```diff
@@ -46,6 +46,32 @@ interface SnowflakePluginInput {
     storage: StorageExtension
 }
 
+/**
+ * Util function to return a promise which is resolved in provided milliseconds
+ */
+function waitFor(millSeconds) {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      resolve();
+    }, millSeconds);
+  });
+}
+
+async function retryPromiseWithDelay(promise, nthTry, delayTime) {
+  try {
+    const res = await promise;
+    return res;
+  } catch (e) {
+    if (nthTry === 1) {
```

Comment:
nit: I unintentionally created infinite loops in the past by messing up the decrement, so I always write this kind of condition as a "less than":

```suggestion
    if (nthTry <= 1) {
```